site/app/libraries/database/DatabaseQueries.php

```diff
@@ -452,6 +452,55 @@ public function getCategoriesIdForThread($thread_id) {
         return $categories_list;
     }
 
+    public function splitPost($post_id, $title, $categories_ids) {
+        $this->course_db->query("SELECT * from posts where id = ?", array($post_id));
+        $posts = $this->course_db->rows();
+        $old_thread_id = -1;
+        $thread_id = -1;
+        // Safety measure in case the database is bad for some reason
+        $counted_posts = array();
+        if (count($posts) > 0) {
+            $post = $posts[0];
+            if ($post["parent_id"] != -1) {
+                $old_thread_id = $post["thread_id"];
+                $this->course_db->query("SELECT id from threads where merged_post_id = ?", array($post_id));
+                $thread_id = $this->course_db->rows();
+                if (count($thread_id) > 0) {
+                    $thread_id = $thread_id[0]["id"];
+                    $this->course_db->query("UPDATE threads set merged_thread_id=-1, merged_post_id=-1 where id=?", array($thread_id));
+                    $this->course_db->query("DELETE FROM thread_categories where thread_id=?", array($thread_id));
+                }
+                else {
+                    $this->course_db->query("INSERT INTO threads (title, created_by, is_visible, lock_thread_date) VALUES (?, ?, ?, ?)", array($title, $post["author_user_id"], true, null));
+                    $this->course_db->query("SELECT MAX(id) as max_id from threads where title=? and created_by=?", array($title, $post["author_user_id"]));
```

Comment:
Changing the code in AbstractDatabase.php to always run the code under Utils::startsWith($lower, 'select') will get the `returning` syntax to work, but I won't be pushing that change since this is used by a lot of things (so if it was changed it may be better to make a new commit) and I don't know enough about how the database works to understand if there would be any issues that result from this.

Reply:
It may be better to use a comment like: 
```
// TODO: Update AbstractDatabase.php to work with returning syntax 
```
rather than commented out code.